projects/igniteui-angular/migrations/common/tsUtils.ts

```diff
@@ -57,3 +60,99 @@ export function getImportModulePositions(sourceText: string, startsWith: string)
+    const positions = [];
+
+    const imports = source.statements.filter(<(a: ts.Statement) => a is ts.ImportDeclaration>namedImportFilter);
+    if (!imports.length) {
+        return positions;
+    }
+    const elements: ts.NodeArray<ts.ImportSpecifier> =
+        imports
+        .map(x => (x.importClause.namedBindings as ts.NamedImports).elements)
+        .reduce((prev, current) => prev.concat(current) as unknown as ts.NodeArray<ts.ImportSpecifier>);
+
+    for (const elem of elements) {
+        if (elem.propertyName && elem.propertyName.text === name) {
+            // alias imports `igxClass as smth` -> <propertyName> as <name>
+            // other references are only for the name portion
+            positions.push({ start: elem.propertyName.getStart(), end: elem.propertyName.getEnd() });
+            break;
+        }
+
+        if (!elem.propertyName && elem.name.text === name) {
+            const renames = service.findRenameLocations(sourcePath, elem.name.getStart(), false, false, false);
+            if (renames) {
+                const renamesPos = renames.map(x => ({ start: x.textSpan.start, end: x.textSpan.start + x.textSpan.length}) );
+                positions.push(...renamesPos);
+            }
+        }
+    }
+
+    return positions;
+}
+
+//#region Language Service
+
+/**
+ * Create a TypeScript language service
+ * @param filePaths Paths for files to include for the language service host
+ * @param host Virtual FS host
+ * @param options Typescript compiler options for the service
+ */
+export function getLanguageService(filePaths: string[], host: Tree, options: ts.CompilerOptions = {}): ts.LanguageService {
+    // https://stackoverflow.com/a/14221483
+    const fileVersions = new Map<string, number>();
+    patchHostOverwrite(host, fileVersions);
+
+    const servicesHost: ts.LanguageServiceHost = {
+        getCompilationSettings: () => options,
+        getScriptFileNames: () => filePaths,
+        getScriptVersion: fileName => {
+            // return host.actions.filter(x => x.path === fileName && x.kind !== 'c').length.toString();
+            const version = fileVersions.get(fileName) || 0;
+            return version.toString();
+        },
+        getScriptSnapshot: fileName => {
+            if (!host.exists(fileName)) {
+                return undefined;
+            }
+            return ts.ScriptSnapshot.fromString(host.read(fileName).toString());
+        },
+        getCurrentDirectory: () => process.cwd(),
+        getDefaultLibFileName: opts => ts.getDefaultLibFilePath(opts),
+        fileExists: fileName => {
+            return filePaths.indexOf(fileName) !== -1;
+        }
+    };
+    return ts.createLanguageService(servicesHost, ts.createDocumentRegistry());
+}
+
+function patchHostOverwrite(host: Tree, fileVersions: Map<string, number>) {
+    const original = host.overwrite;
+    host.overwrite = (path: string, content: Buffer | string) => {
+        if (!fileVersions.has(path)) {
```

Comment:
Very nit picky, especially after the black magic you're doing above:
Since `fileVersions` are entirely handled here, you can do something like:
```
const version = fileVersions.get(path) || 0;
if (!version) {
...
} else {
fileVersions.set(path, version + 1);
}
```
Just to avoid having to do two operations on the `Map`